adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -550,11 +568,26 @@ public OrderData handle3DResponse(final HttpServletRequest request) throws Excep
                 throw new SignatureException("MD does not match!");
             }
 
-            restoreSessionCart();
             PaymentsResponse paymentsResponse = getAdyenPaymentService().authorise3DPayment(sessionPaymentData, paRes, md);
+
+            String orderCode = paymentsResponse.getMerchantReference();
+            OrderModel orderModel = retrieveOrder(orderCode);
+
+            updateOrderPaymentInfo(paymentsResponse, orderModel);
+
```

Comment:
We can move this to later stage where we are setting order status to make sure we are updating all order related details at one place.

Reply:
Makes sense, will do